dev/integration_tests/native_driver_test/lib/display_cutout_rotation_main.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_driver/driver_extension.dart';
+import 'package:native_driver/extension.dart';
+
+import 'src/allow_list_devices.dart';
+
+void main() async {
+  ensureAndroidOrIosDevice();
+  enableFlutterDriverExtension(commands: <CommandExtension>[
+    nativeDriverCommands,
+  ]);
+
+  // Run on full screen.
+  await SystemChrome.setEnabledSystemUIMode(SystemUiMode.immersive);
+  runApp(const MainApp());
+}
+
+final class MainApp extends StatefulWidget {
+  const MainApp({super.key});
+
+  @override
+  State<MainApp> createState() => _MainAppState();
+}
+
+class _MainAppState extends State<MainApp> {
+  @override
+  Widget build(BuildContext context) {
```

Comment:
Please leave comments explaining what this is doing, it's really hard to understand where (or what) this does and exactly what we're checking for.